vm/src/pyobjectrc.rs

```diff
@@ -0,0 +1,188 @@
+use crate::common::rc::{PyRc, PyWeak};
+use crate::pyobject::{IdProtocol, PyObject, PyObjectPayload};
+use std::borrow;
+use std::fmt;
+use std::ops::Deref;
+
+pub struct PyObjectRc<T = dyn PyObjectPayload>
+where
+    T: ?Sized + PyObjectPayload,
+    PyRc<PyObject<T>>: AsPyObjectRef,
+{
+    inner: PyRc<PyObject<T>>,
+}
+
+pub type PyObjectWeak<T = dyn PyObjectPayload> = PyWeak<PyObject<T>>;
```

Comment:
Should probably be a newtype as well :grimacing: still the issue of downgrade->upgrade to Arc->drop original PyObjectRc->drop Arc without `__del__`. It would also let you add the upgrade() method to PyObjectWeak and have it return a PyObjectRc